### PR TITLE
Feature/editions

### PR DESCRIPTION
**Pull request by mattrout92**
### What

Acceptance tests for editions endpoints on code list api

### How to review

Run the code list api (make acceptance) on feature/editions
Run the api tests
`cd codeListAPI`
`go test ./...`

### Who can review

Anyone

### CODE REVIEW

File: codeListAPI/get_edition_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	if err := ds.SetupCodelistEditions(); err != nil {
+		t.Error(err)
+		t.Fail()
+	}
+
+	codelistAPI := httpexpect.New(t, cfg.CodeListAPIURL)
+
+	Convey("Given an edition of a codelist eqists", t, func() {
```
Comment (eldeal) [inline codeListAPI/get_edition_test.go]: exists doesn't have a q in it :)

File: codeListAPI/get_edition_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		Convey("When a request is made for the edition", func() {
+			Convey("Then the codelist is returned to the user", func() {
+				response := codelistAPI.GET("/code-lists/ABCDEF/editions/2018").Expect().Status(http.StatusOK).JSON().Object()
+
+				response.Value("id").Equal("ABCDEF")
```
Comment (eldeal) [inline codeListAPI/get_edition_test.go]: I'm not certain we should be returning ID as the code list name here. not sure ID is a meaningful field for this endpoint.

File: codeListAPI/get_edition_test.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		})
+	})
+
+	Convey("Given a user wants to know the edition of a code list that doesn't exist", t, func() {
+		Convey("When a request is made for the edition", func() {
+			Convey("Then the a not found status is returned to the user", func() {
+
+				codelistAPI.GET("/code-lists/BCDSOE/editions/2018").Expect().Status(http.StatusNotFound)
+
+				codelistAPI.GET("/code-lists/ABCDEF/editions/2019").Expect().Status(http.StatusNotFound)
```
Comment (eldeal) [inline codeListAPI/get_edition_test.go]: Feels like 'an edition of a code list which doesnt exist' and 'an edition which doesnt exist of a code list that does' should be different test cases to more easily determine possible failures in future

File: codeListAPI/get_editions_test.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	if err := ds.SetupCodelistEditions(); err != nil {
+		t.Error(err)
+		t.Fail()
+	}
+
+	codelistAPI := httpexpect.New(t, cfg.CodeListAPIURL)
+
+	Convey("Given two editions of the same codelist exists", t, func() {
+		Convey("When a request is made for the editions", func() {
+			Convey("Then the codelists are returned to the user", func() {
```
Comment (eldeal) [inline codeListAPI/get_editions_test.go]: the editions are returned to the user, rather than codelists, as there's only one list at this point?

**Review by eldeal — CHANGES_REQUESTED** [review]

**Review by CarlHembrough — APPROVED** [review]
LGTM - we will need to revisit this when the data model is updated

**Pull request merged**